src/main/java/org/sagebionetworks/bridge/services/AdminAccountService.java

```diff
@@ -0,0 +1,321 @@
+            try {
+                PasswordAlgorithm passwordAlgorithm = DEFAULT_PASSWORD_ALGORITHM;
+                String passwordHash = passwordAlgorithm.generateHash(submittedAccount.getPassword());
+                account.setPasswordAlgorithm(passwordAlgorithm);
+                account.setPasswordHash(passwordHash);
+                account.setPassword(null);
+            } catch (InvalidKeyException | InvalidKeySpecException | NoSuchAlgorithmException ex) {
+                throw new BridgeServiceException("Error creating password: " + ex.getMessage(), ex);
+            }
+        }
+        Set<Roles> finalRoles = updateRoles(context, submittedAccount.getRoles(), account.getRoles());
+        account.setRoles(finalRoles);
+
+        accountDao.createAccount(account);
+        
+        sendVerificationMessages(app, account);
+        return account;
+    }
+    
+    public Account updateAccount(String appId, Account account) {
+        checkNotNull(account);
+        
+        App app = appService.getApp(appId);
+        if (app == null) {
+            throw new EntityNotFoundException(App.class);
+        }
+        if (account.getId() == null) {
+            throw new EntityNotFoundException(Account.class);
+        }
+        account.setPassword(null); // don’t validate this value
+
+        AccountId accountId = AccountId.forId(appId,  account.getId());
+        Account persistedAccount = accountDao.getAccount(accountId)
+                .orElseThrow(() -> new EntityNotFoundException(Account.class));
+        if (!persistedAccount.isAdmin()) {
+            throw new UnauthorizedException(CALLER_NOT_ADMIN_MSG);
+        }
+
+        Validator validator = new AdminAccountValidator(app.getPasswordPolicy(), app.getUserProfileAttributes());
+        Validate.entityThrowingException(validator, account);
+        
+        // None of these values should be changeable by the user.
+        account.setAppId(persistedAccount.getAppId());
+        account.setAdmin(persistedAccount.isAdmin());
+        account.setDataGroups(persistedAccount.getDataGroups());
+        account.setCreatedOn(persistedAccount.getCreatedOn());
+        account.setHealthCode(persistedAccount.getHealthCode());
+        account.setPasswordAlgorithm(persistedAccount.getPasswordAlgorithm());
+        account.setPasswordHash(persistedAccount.getPasswordHash());
+        account.setPasswordModifiedOn(persistedAccount.getPasswordModifiedOn());
+        account.setOrgMembership(persistedAccount.getOrgMembership());
+        account.setMigrationVersion(persistedAccount.getMigrationVersion());
+        account.setDataGroups(addToSet(account.getDataGroups(), TEST_USER_GROUP));
+        account.setModifiedOn(getModifiedOn());
+        account.setPhoneVerified(persistedAccount.getPhoneVerified());
+        account.setEmailVerified(persistedAccount.getEmailVerified());
+        if (!RequestContext.get().isInRole(ADMIN, WORKER)) {
```

Comment:
Need to be able to add email and phone and change them.

Reply:
No this is correct, account is the submitted values and you can change email/phone (they are not copied over from the persisted account)